datafusion/functions/src/core/mod.rs

```diff
@@ -81,10 +80,6 @@ pub mod expr_fn {
         coalesce,
         "Returns `coalesce(args...)`, which evaluates to the value of the first expr which is not NULL",
         args,
-    ),(
-        make_map,
-        "Returns a map created from the given keys and values pairs. This function isn't efficient for large maps. Use the `map` function instead.",
-        args,
```

Comment:
I'm not sure where can put this doc. Maybe we can do it when https://github.com/apache/datafusion/issues/11435

Reply:
Agreed, We can use to document this function in https://datafusion.apache.org/user-guide/sql/scalar_functions.html